src/course4/homework/Challenge5b.java

```diff
@@ -12,21 +12,16 @@ public static void main(String[] args) {
 
         System.out.println("Introduce your number:");
         double number = scan.nextDouble();
-        boolean trueValue = true;
 
-        // next we have two "if" loops to verify if your number is higher, lower or equal to 0
-        if (number != 0) { //this verifies if the number is different than 0;
-
-            trueValue = (number > 0); //next step if the number is different than 0 (I choose higher out of habit;
-                                      // it can also be lower;
+        //we have two "if" loops to verify if your number is higher, lower or equal to 0
+        if (number == 0) { //this verifies if the number is 0;
+            System.out.println("Use a number higher or lower than 0.");
 
-            if (trueValue) { //next step if the number is higher than 0;
-                System.out.println("Your number is a positive number.");
+            if (!(number < 0)) { //next step if the number is higher than 0 (the "true" or "false" result in paranthesis is reversed);
```

Comment:
this condition in never satisfied because at this point your number is 0 (you are inside above `if` from line 17). You can test the program by entering a positive number from keyboard and see that nothing is displayed. 
But, again, I don't understand why you over-complicate so much with things we didn't study yet. `Challenge5` would be enough for now.

Reply:
I've modified the code and went outside the "if" from line 17 and now it works as it should. Also, the double negation from line 17 is that I use the "!" operator like the challenge requested and write my condition for the number to be equal to 0 by negating it being different than 0.